app/views/simulation.scala.html

```diff
@@ -109,179 +53,18 @@
     <script type="text/nlogo" id="nlogo-code" data-filename="model.nlogo"></script>
 
     <script>
+      window.debugMode = @{environment.mode != play.api.Mode.Prod};
+      Ractive.DEBUG = window.debugMode
+      window.standaloneURL = `${window.location.protocol}@TagBuilder.protocolRelativeURL(
+        new java.net.URL(routes.Local.standalone.absoluteURL)
+      )`;
+    </script>
```

Comment:
What's the reasoning for doing it this way?  Why do we not export `standaloneURL` when we define it?  Even if there's a good reason why we don't, it might be nice to have a comment in code here, explaining why this (and almost only this) is exempt from the module system.

If it *is* reasonable to put `standaloneURL` into a module, then `debugMode` could also probably be exported.  They could be exported from a hypothetical inline `<script type="module" name="/simulationglobals.js">` here (or similar name).

Reply:
The reason these variables cannot be defined in a separate script file is they come from the Play view (`@standalone` etc.).

A `<script type="module" name="simulationglobals.js">` would be really nice, but unfortunately [the `name` attribute doesn't exist](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attributes) and there currently is no other builtin way of adding a filename to an inline script. (Technically except messing with the way browser modules work using a service worker, but that's not something I would do).

I added comments explaining why these globals exist to the code.

Reply:
Ah, of course!  That makes sense. :sunglasses: :+1: